### DISCUSSION TITLE
How to remove authenticator app

### DISCUSSION THREAD

**Discussion by jheidbrink** [community/community#54699] [New to GitHub]
I configured a TOTP authenticator app. Later i also configured a security key which was automatically selected as the preferred 2FA method. Now i want to remove the TOTP authenticator so that it no longer shows up as configured, and so that I cannot accidentally select it as preferred method. I want to remove the corresponding secret from my app then. However in the Web GUI I don't find an option to remove the configured authenticator app.

**Comment by iamreiyn** [comment]
Hello, @jheidbrink 

Unfortunately, as of now, there is no way to remove TOTP (Authenticator app) directly. However, **you can remove it by signing up for the 2FA again using the SMS/Text message method.**

So, you must either have the Authenticator app or SMS/Text method configured. This way you can reconfigure to make your security key as the preferred method and also have the Authenticator app removed.

And in order to do that:

1. Open your GitHub **Settings**

![procedure](https://i.postimg.cc/mkjzYZwF/Screenshot-20230508-051329-Edge.jpg)

2. Navigate to **Password and authentication**

![procedure](https://i.postimg.cc/9XY751hF/Smart-Select-20230508-052308-Edge.jpg)

3. Click on the **Enabled** button next to "Two factor authentication" (it will disable 2FA)

After it is disabled, you can enable it again using the SMS/text option and then reconfigure your Security Key and make it preferred.

Thank you!

**Reply by lorisleiva** [reply]
Unfortunately, this workaround no longer works for some users as we cannot disable 2FA in order to reconfigure it.

<img width="803" alt="CleanShot 2023-09-15 at 17 50 53@2x" src="https://github.com/community/community/assets/3642397/173cadff-7d0a-4ee8-939d-ba88f445de28">

**Comment by jheidbrink** [comment]
Thank you @montymahato 

I consider 2FA via SMS less secure than via Yubikey. So instead of the above, I now configured my Yubikey to not only act as security key, but also as TOTP generator. Thus I have effectively limited 2FA to Yubikey only.

**Reply by iamreiyn** [reply]
Anytime, glad to know that you were able to work your way around it!

**Reply by FlameDuck** [reply]
Of course 2FA sent unecrypted over the air using a protocol hacked together by GSM in the 1980's is less secure than an encrypted purpose built hardware device from 2018. Anyone who thinks differently is simply wrong.

**Comment by DNin01** [comment]
> I consider 2FA via SMS less secure than via Yubikey. So instead of the above, I now configured my Yubikey to not only act as security key, but also as TOTP generator. Thus I have effectively limited 2FA to Yubikey only.

That's cool that you found a workaround.

GitHub did say in a blog post that [they were working on making it possible to use security keys as the primary second factor](https://github.blog/changelog/2022-11-21-updates-to-the-two-factor-authentication-setup-flow/), so if I'm right, this option will be available soon.

> In the future, you'll also find security keys there as an option for initial setup on supported devices and browsers.

**Comment by tianhuil** [comment]
Agree with @jheidbrink.  But because I use to use the authenticator app prior to getting Yubi keys, it's now stuck there and I can't remove it.  The only workaround is "editing" the authenticator app password, confirming, and then immediately deleting the key from my phone.

I wish they would just let me delete it.

**Reply by n1ngu** [reply]
I was planning on doing exactly this.

But even if I have multiple FIDO keys and backed-up recovery codes, I fear I am bricking the main MFA TOTP method. Plus the attack surface on that method remains open.

**Reply by dr-yd** [reply]
DO NOT DO THIS! Github now forces you to reauthenticate using TOTP 2FA every week even if you have a Yubikey! Completely backasswards, no idea who thought that was a good idea...

**Reply by n1ngu** [reply]
If that is true, this may be my final words with this account until I am asked a TOTP token :vulcan_salute:

**Reply by n1ngu** [reply]
Let's see if I can find that recovery codes backup... :smiling_face_with_tear:

**Comment by l4kr** [comment]
I have 4 damn keys I don't need a damn TOTP. Plz gib option to remove it. @Microsoft

**Reply by pickyusernamepicker** [reply]
While we're at it, 2FA settings in Microsoft accounts are a much worse mess...

**Comment by stevensmedia** [comment]
This is inherently insecure that the user is forced to open up authentication methods that aren't being used. Fix the hole, Microsoft.

**Reply by n1ngu** [reply]
Indeed FIDO keys are secure enough. Additional MFA methods only increase the attack surface on accounts.

**Reply by mikaellq** [reply]
It seems rather meaningless to use security keys if authenticator app can’t be removed.
Then the account is no more secure than with authenticator app only. 👎

**Comment by stevensmedia** [comment]
It's worse now. It demands you give an authenticator code EVEN IF YOU DON'T HAVE AN AUTHENTICATOR.

I have disabled TFA and will not be re-enabling it until this is fixed. I have a support ticket in insisting this be fixed.

I recommend project maintainers prepare to move active development OFF of github, so anyone using the github version is at risk of using older, insecure versions of software until this is fixed.

**Comment by wwangsa** [comment]
Prior to setting up Yubi Key and authenticator app, I used Github Mobile. Now, I can't remove the authenticator app. I'm able to login with the authenticator app but when I go to settings > password and authentication > authenticator app > "Verify the code from the app" , I get "Two-factor code verification failed. Please try again.". It looks like it doesn't recognize the authenticator app but I still can use it at login...

**Reply by horror-proton** [reply]
I think that "Verify the code from the app" is only used to check if a new authenticator app is properly configured via the QR code above, not existing ones.

**Comment by pmalek** [comment]
I don't understand why we're not able to remove a particular method of 2FA given that we still have other methods set up. 🤔 

<img width="1034" alt="image" src="https://github.com/community/community/assets/739996/9a0f4706-28c9-4ca3-98f5-33478d9c3a06">

**Comment by lltr** [comment]
I am also wondering why the option to remove the Authenticator app is not available once Security keys are configured. The Authenticator app is not phishing resistant. It is counter intuitive to me for 2FA to have another surface open and I consider this to be less secure. I hope the option to remove the Authenticator app will be available soon.

**Reply by pickyusernamepicker** [reply]
The option to disable the recovery codes would be sensible too. Google's already added it.

**Comment by jsanadev** [comment]
You can disable 2FA using your password.

**Reply by l4kr** [reply]
There is no way you are serious and just ignored what the whole conversation is about...

**Reply by jsanadev** [reply]
i just did it, i disabled 2FA to reconfigure authenticator app, it just prompt you requesting authenticator code but under it, it shows you "disable using your password"

**Reply by l4kr** [reply]
We are not talking about disabling 2FA. We are talking about disabling TOTP when we have passkeys set up.

**Reply by jsanadev** [reply]
I just followed the title from post, _How to remove authenticator app_, apologies if i was wrong, i was trying to help with that use case

**Reply by kr** [reply]
It's a reasonable misunderstanding, the phrasing is arguably ambiguous. In this context, "authenticator app" means TOTP specifically, it doesn't refer to other forms of 2FA.

It would probably be clearer for a lot of people if the title could be changed to explicitly say TOTP somewhere.

**Reply by tasmo** [reply]
Not anymore. Github enforced to use Two-Factor for many accounts.

**Comment by wrinkledeth** [comment]
It's 2024 now, looks like there is still no option to disable the "Authenticator App" option if you have security keys setup: 
<img width="912" alt="image" src="https://github.com/community/community/assets/97406595/36863656-cf34-4c04-85aa-72aa90e2bc89">

Any plans of getting this fixed?

**Reply by fengyi233** [reply]
Please check here. 
![CC8E9CFA7DDA802F848824CC650ADF89](https://github.com/user-attachments/assets/7a074403-79c1-4671-a0c9-82949381fb3d)

**Reply by pickyusernamepicker** [reply]
@fengyi233 That option disables all 2FA, including hardware keys. This thread is about keeping hardware keys enabled & disabling only the app (TOTP apps offer less security than hardware keys).

**Reply by tasmo** [reply]
This is not possible anymore. Two-factor authentication is enforced on most accounts.
<img width="988" alt="Screenshot 2024-09-11 at 13 07 25" src="https://github.com/user-attachments/assets/c080dfd5-2d85-45dc-8ee6-b24e7e77195e">

**Comment by knan** [comment]
With multiple hardware keys setup, totp app is superfluous and a hindrance. Please let us remove it.

**Comment by stanleyguevara** [comment]
I just managed to remove it.

Just set up a `SMS/Text message` method, authenticate with it and then you can disable `Authenticator app` option.

I had to do it because of Raivo iOS app getting acquired and losing everybody's keys on first update. Do not recommend. Lucky I had my recovery codes.

**Reply by wrinkledeth** [reply]
Were you able to disable SMS auth method afterwards?

**Reply by stanleyguevara** [reply]
Yes. But only after setting up a new OTP authenticator app.
Looks like it requires backup codes to be set up + 2 independent 2FA methods.

**Comment by ghostinhershell** [comment]
Hey there everyone 👋🏾 

Just wanted to circle back here and give an updated answer to this question since I think it's still a popular question in this community. 

I'm afraid that it is not possible to remove a TOTP application from an account enabled for 2FA. GitHub requires that a primary factor - either TOTP or SMS - be active for all accounts and so the only option is to replace one with another. Any additional access factors can be setup (or removed) at your discretion and can be used for day-to-day access; however the primary factor must remain active at all times. You can learn more here: [Two Factor Authentication Frequently Asked Questions 🔑 🗝️](https://github.com/orgs/community/discussions/68240).
 
Apologies that we cannot accommodate your request at this time though I will pass on your comments to our authentication team.  I can't promise any changes will be made but we will consider all feedback as we iterate on this, and future security-related features. If there's anything additional you might like to add though, or if I can help with anything else, then please do let me know.

**Reply by n1ngu** [reply]
Thanks for the update!

Could you please clarify what are the consequences of loosing this "primary factor" as long as "additional access factors" are operative? E.g. could you shed some light on this thread https://github.com/orgs/community/discussions/54699#discussioncomment-6015764

Also, when reaching the authentication team, please bring to their attention that excessive access factors only increase the attack surface on github accounts https://github.com/orgs/community/discussions/54699#discussioncomment-6824929

**Reply by ngist** [reply]
It's strange that TOTP would be a required Auth factor, when HW security keys using FIDO2/webauthn should be inherently more secure and at least as available as TOTP.

I just setup redundant HW security keys on my account, and was surprised to find TOTP can't be removed. 

TOTP is readily phishable, and once in the account an attacker can just add more auth factors so it's not supplying added security, and opens up a phishing attack vulnerability that other auth methods don't have. You could say well just don't use TOTP if you don't want to, but a user could still be tricked into thinking they need to supply a TOTP code since if it can't be removed it must be needed for something right? I could just delete the TOTP form my auth app, but I'm not going to because if I can't eliminate it from githubs side, I might need it one day.

**Reply by torburg** [reply]
Hi there!

I'd like to share my experience with GitHub's two-factor authentication options. Currently, I've set up GitHub Mobile as my primary 2FA method, as it's listed as the recommended option on the site.

However, I've noticed something puzzling - even though GitHub Mobile is configured as my primary 2FA method, the system still asks me for an authentication code from an authenticator app when I log in.

I've been trying to use 1Password (an older version) as my authenticator app, but the codes it generates aren't working for GitHub. I've even tried re-linking my account in the app, but that didn't solve the issue.

It would be really helpful if GitHub could make their own mobile app authentication work seamlessly as the primary method when selected. This would eliminate the need for users to download and configure additional authentication apps for the same purpose.

Would it be possible to improve this experience so that when users select GitHub Mobile as their primary 2FA method, the system respects that choice?

Thanks for considering this feedback!

**Reply by AVargas-84** [reply]
I'm having the same isssue with the primary 2FA method. I first configured an authenticator app, but then switched phones and started using github app. Now it always asks for the authenticator app, even though i have the prefered method selected for the github app. I don't have the authenticator app anymore, and I prefer to use the github app instead of some third party authenticator. It would be a great solution to be able to deactivate one and select the other. Thanks.

**Comment by happymoodhub** [comment]
- Login and go to your 2FA/security settings
- Temporarily disable all 2FA methods
- Re-enable only your security key (leave TOTP unconfigured)
- Now the TOTP entry won’t show up again
- if problem not solved, download an authenticator app from [trusted platform](https://www.happymoodhub.com/)

**Reply by escape0707** [reply]
If you've read posts in this thread:

Upon your suggested second step, a notice saying:

> "Because of your contributions on GitHub, two-factor authentication is required for your account. Thank you for helping keep the ecosystem safe! [Learn more about our two-factor authentication initiative](https://github.blog/2023-03-09-raising-the-bar-for-software-security-github-2fa-begins-march-13/)."

will show at the 2FA section of the settings page, and you can't just disable 2FA for that GitHub account. That's why many people are complaining here in the first place.

**Comment by azu** [comment]
A part of the MFA improvements, I realized that the Authenticator app couldn't be erased, so we got here.

Recent phishing attacks have been carried out using TOTP replay attack.
Therefore, I want to disable the Authenticator app itself and just have security keys.

- Specific example of the attack: [npm debug and chalk packages compromised](https://www.aikido.dev/blog/npm-debug-and-chalk-packages-compromised) (the victim used TOTP as MFA)

npm was able to remove the Authenticator app, so I'm surprised that GitHub can't remove the Authenticator app

**Comment by pacaroha** [comment]
Today I logged into this account, which is my public work identity for GitHub with my current employer. I have used a YubiKey for various things at this job as long as I had one, and figured I'd set it up for this GitHub account after the deadline forcing everyone to enable 2FA hit.

Except now I've been forced into creating a TOTP key that I don't want or need, and now cannot delete.

It's been years now since we were told that we would "soon" be able to set up a security key as a 2FA method in the initial flow. This is ridiculous.

**Comment by cunhaax** [comment]
2026... how I got here?

Have the following in 2FA:

* Backup codes
* Authenticator app
* Github mobile
* 2 security keys!

But Github doesn't allow me to remove the Authenticator App.
Make this make sense. 🤡🤡🤡

**Comment by booi** [comment]
Q2 2026 checking in. How is this still open? I asked AI (not copilot) how do I disable TOTP on github and it genuinely appeared flabbergasted that it seems impossible. The fact that SMS is still even offered blows my mind. 

It appears that github (especially after the latest hack) doesn't seem to be willing to evolve with the world and the ever increasing importance of security.

**Comment by echohack** [comment]
Hi GitHub folks, please allow us to remove TOTP if we have a security key configured. Thank you!

**Comment by Gecko51** [comment]
Still no native fix as of mid-2026, based on what others are reporting in this thread and across similar discussions.

The current situation: GitHub requires a primary 2FA factor (TOTP or SMS) to always be configured. Security keys and passkeys are treated as supplemental factors, not primary ones, which is why TOTP removal is blocked even when you have multiple hardware keys set up.

The options available right now:

1. Delete the TOTP secret from your authenticator app. The entry stays visible in GitHub settings but without the secret on your device, TOTP effectively becomes unusable for an attacker who does not have your phone. Your security key remains your working day-to-day method. This is what most people in this situation end up doing.

2. Replace TOTP with SMS, then reconfigure. Some users have reported success adding SMS as a 2FA method, which then allows switching the primary factor. After SMS is active, TOTP may become removable. This path depends on whether your account has 2FA enforcement active -- enforced accounts may not let you disable or reconfigure at all.

3. Configure your hardware key as a TOTP generator as well (e.g. YubiKey supports this via its OATH functionality). You would then delete the software authenticator secret and keep TOTP tied exclusively to the physical key.

None of these are ideal. The underlying issue -- GitHub not allowing TOTP removal when stronger factors are present -- is a genuine security regression that a lot of people here have flagged. Hopefully the authentication team addresses it, but there has been no public commitment to do so yet.

**Reply by booi** [reply]
i honestly don't know if anyone works on github product anymore. They lack a leader (https://github.com/about/leadership) and seem utterly uninterested in the waves of projects looking for new homes.

**Comment by eventsourcer** [comment]
Following on that, have they added any workaround to remove ms authenticator while other options are configured?
This is completely nonsense and should have been fixed by now.

**Comment by afk11** [comment]
Added an authenticator TOTP token just to test something, and now I'm stuck with these attached to my account. This is absurd.
